### PR TITLE
Add condition support to WithConstructionBeamOverlay.

### PR DESCRIPTION
**Pull request by GraionDilach**

### CODE REVIEW

**Comment by Mailaender** [comment]
Thanks. Seems I forgot to push that change. I also didn't use invalid token properly in my variant.

**Comment by GraionDilach** [comment]
You're welcome then. 😄

File: OpenRA.Mods.HV/Traits/Render/WithConstructionBeamOverlay.cs [hunk 1]
```diff
@@ -58,9 +64,20 @@ public WithConstructionBeamOverlay(ActorInitializer init, WithConstructionBeamOv
...
+				token = self.GrantCondition(info.Condition);
+
 			// Remove the animation once it is complete
 			if (overlay != null && renderSprites != null && anim != null)
-				overlay.PlayThen(info.Sequence, () => self.World.AddFrameEndTask(w => renderSprites.Remove(anim)));
+			{
+				overlay.PlayThen(info.Sequence, () => self.World.AddFrameEndTask(w =>
+				{
+					if (token != Actor.InvalidConditionToken)
+						token = self.RevokeCondition(token);
```
Comment (Mailaender) [inline OpenRA.Mods.HV/Traits/Render/WithConstructionBeamOverlay.cs:76]: Actually this is wrong. I had the reverse in mind.

Comment (Mailaender) [reply OpenRA.Mods.HV/Traits/Render/WithConstructionBeamOverlay.cs:76]: Yours makes more sense though.

**Pull request merged**